src/module_compilation.ml

```diff
@@ -37,22 +37,25 @@ let build_cm cctx ?sandbox ?(dynlink=true) ~dep_graphs ~cm_kind (m : Module.t) =
   let obj_dir  = CC.obj_dir       cctx in
   let ctx      = SC.context       sctx in
   let private_obj_dir = CC.private_obj_dir cctx in
+  let vlib_modules = CC.vlib_modules cctx in
   Option.iter (Mode.of_cm_kind cm_kind |> Context.compiler ctx) ~f:(fun compiler ->
     Option.iter (Module.cm_source m cm_kind) ~f:(fun src ->
       let ml_kind = Cm_kind.source cm_kind in
       let dst = Module.cm_file_unsafe m ~obj_dir cm_kind in
       let extra_args, extra_deps, other_targets =
-        match cm_kind, m.intf with
+        match cm_kind, m.intf
+              , Module.Name.Map.find vlib_modules (Module.name m) with
```

Comment:
Since we are only matching on the module being virtual or not, we can use `Map.mem`.